### PR TITLE
feat: add deployment metadata to the table of the deployment frequency details dashboard

### LINKED ISSUE #7280: [Feature][Dashboard] Add deployment metadata to the table of the 'DORA details - deployment frequency' dashboard
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

When diving into the 'DORA details - deployment frequency' dashboard. I found it hard to associate the deployment id with the actual deployment in the GitHub UI. 

image

It would be better if a readable name and URL of the deployment could be displayed in this table.

### Description

Add two columns to the above table:
1. `Name`: A readable name of the deployment. Take GitHub workflow run for example, it could 
   - The title of the PR that triggers the deployment or workflow run, e.g. https://github.com/apache/incubator-devlake/actions/runs/8538161970
   - Or, the message of the commit that triggers the deployment or workflow run, e.g. https://github.com/apache/incubator-devlake/actions/runs/8505543174

   Please refer to the GitHub deployment list or workflow run list for more details: https://github.com/apache/incubator-devlake/actions?page=9 

   - deployment list
   image

   - workflow run list
   image

The naming logic should be similar for GitLab/BitBucket/Azure/Bamboo deployment or pipeline runs.

2. A `url` for users to visit. For instance, 
- if a deployment is transformed from a GitHub deployment, the URL would be the GitHub deployment's address, e.g. https://github.com/apache/seatunnel/deployments
- if a deployment is transformed from a GitHub workflow run, the URL would be the GitHub workflow run's address, e.g. https://github.com/apache/incubator-devlake/actions/runs/8538161970

### Related issues

#dora drill

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by abeizn**
### Summary
feat: add deployment metadata to the table of the dloyment frequency details dashboard

### Does this close any open issues?
Closes #7280 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/gitlab/models/pipeline.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ type GitlabPipelineProject struct {
 	PipelineId      int    `gorm:"primaryKey"`
 	ProjectId       int    `gorm:"primaryKey"`
 	Ref             string `gorm:"type:varchar(255)"`
+	WebUrl          string `gorm:"type:varchar(255)"`
```
Comment (d4x1) [inline backend/plugins/gitlab/models/pipeline.go:61]: varchar(255) maybe too short.

**Review by d4x1 — DISMISSED** [review]

File: backend/core/models/domainlayer/devops/cicd_deployment.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ type CICDDeployment struct {
 	domainlayer.DomainEntity
 	CicdScopeId         string `gorm:"index;type:varchar(255)"`
 	Name                string `gorm:"type:varchar(255)"`
+	DisplayTitle        string
```
[the remark this answers is not in the document]
Comment (abeizn) [reply backend/core/models/domainlayer/devops/cicd_deployment.go:28]: longtext

File: backend/plugins/gitlab/models/pipeline.go (hunk 1 above)
Comment (abeizn) [reply backend/plugins/gitlab/models/pipeline.go]: deleted

**Review by Startrekzky — APPROVED** [review]
LGTM

**Pull request merged**